File: 455. Assign Cookies/solution.go
import "sort"

func findContentChildren(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)

	var j, flag int
	for i := 0; i < len(g); i++ {
		for j < len(s) && s[j] < g[i] {
			j++
		}

		if j >= len(s) {
			break
		} else {
			flag++
			j++
		}
	}
	return flag
}

File: 455. Assign Cookies/solution.py
class Solution:
    def findContentChildren(self, chil: List[int], cook: List[int]) -> int:
        i = j = res = 0
        chil.sort(reverse=True)
        cook.sort(reverse=True)

        while i < len(chil) and j < len(cook):
            if cook[j] >= chil[i]:
                res += 1
                j += 1
            i += 1

        return res
